src/base-xhr.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+var objectMirrors = require('./object-mirrors');
+
+var BaseXHR = function(xhrImpl, onreadystatechange, onprogress, onloadend) {
```

Comment:
the difference between these props and the other ones is that i want to trigger these handlers myself in shaper instead of the XHR impl triggering them.